pkg/block/metadata/meta.go

```diff
@@ -65,27 +65,6 @@ type ThanosDownsample struct {
 	Resolution int64 `json:"resolution"`
 }
 
-// InjectThanos sets Thanos meta to the block meta JSON and saves it to the disk.
-// NOTE: It should be used after writing any block by any Thanos component, otherwise we will miss crucial metadata.
-func InjectThanos(logger log.Logger, bdir string, meta Thanos, downsampledMeta *tsdb.BlockMeta) (*Meta, error) {
```

Comment:
What's wrong with this? This is really critical to not forget ):

Reply:
Yeah, my bad. I did a robotic job after rebasing on `master`. When I have first started this PR, `InjectThanos` was always called with `downsampledMeta == nil`. Now that you've added proper users, we should revert this change.